packages/suite/src/utils/wallet/coinjoinUtils.ts

```diff
@@ -308,19 +310,31 @@ export const getIsCoinjoinOutOfSync = (selectedAccount: SelectedAccountStatus) =
     }
 };
 
-const roundPhases = [
-    RoundPhase.InputRegistration,
-    RoundPhase.ConnectionConfirmation,
-    RoundPhase.OutputRegistration,
-    RoundPhase.TransactionSigning,
-    RoundPhase.Ended,
-];
+export const getRoundPhaseFromSessionPhase = (sessionPhase: SessionPhase) => {
+    const isValidRoundPhase = (result: number): result is RoundPhase =>
+        Object.values(roundPhases).some(value => value === result);
 
-export const getRoundPhaseFromSessionPhase = (sessionPhase: SessionPhase): RoundPhase =>
-    roundPhases[Number(String(sessionPhase)[0]) - 1];
+    const result = Number(String(sessionPhase)[0]) - 1;
 
-export const getFirstSessionPhaseFromRoundPhase = (roundPhase?: RoundPhase): SessionPhase =>
-    Number(`${(roundPhase || 0) + 1}01`);
+    if (!isValidRoundPhase(result)) {
+        throw new Error(`Invalid round phase value: ${result}`);
+    }
+
+    return result;
+};
+
+export const getFirstSessionPhaseFromRoundPhase = (roundPhase?: RoundPhase) => {
```

Comment:
nit: since you are using the type guards anyways, this is also possible :]

```
const isValidSessionPhase = (phase: number): phase is SessionPhase => {
    const enumValues = Object.values(SessionPhase).slice(Object.keys(SessionPhase).length / 2);

    return enumValues.includes(phase);
};

export const getFirstSessionPhaseFromRoundPhase = (roundPhase?: RoundPhase) => {
    const firstSessionPhase = Number(`${(roundPhase || 0) + 1}1`);

    if (!isValidSessionPhase(firstSessionPhase)) {
        throw new Error(`Invalid round phase: ${roundPhase}`);
    }

    return firstSessionPhase;
};
```

Reply:
Yes... I didn't know that at first. But it is certainly not nice with that `slice`.

Reply:
Not nice 100% but it's a hacky way to streamline a hacky approach – checks out :D The benefit is not having a longer `switch` or smth to decide what the next step is.